tests/temp_tests/test_settings/test_alpsLauncher.py

```diff
@@ -211,8 +216,15 @@ def test_invalid_exclude_hostlist_format():
     ),
 )
 def test_formatting_launch_args(mock_echo_executable, args, expected, test_dir):
-    path, cmd = _as_aprun_command(
-        AprunLaunchArguments(args), mock_echo_executable, test_dir, {}
+    out = os.path.join(test_dir, "out.txt")
+    err = os.path.join(test_dir, "err.txt")
+    open(out, "w"), open(err, "w")
+    shell_launch_cmd = _as_aprun_command(
+        AprunLaunchArguments(args), mock_echo_executable, test_dir, {}, out, err
     )
-    assert tuple(cmd) == expected
-    assert path == test_dir
+    assert isinstance(shell_launch_cmd, ShellLauncherCommand)
+    assert shell_launch_cmd.command_tuple == expected
+    assert shell_launch_cmd.path == pathlib.Path(test_dir)
+    assert shell_launch_cmd.env == {}
+    assert isinstance(shell_launch_cmd.stdout, io.TextIOWrapper)
+    assert isinstance(shell_launch_cmd.stderr, io.TextIOWrapper)
```

Comment:
Might be a  good idea to add this assertion
```py
assert shell_launch_cmd.stdout.name == out
assert shell_launch_cmd.stderr.name == err
```
to make sure the files are pointed at the correct path